./f_check: line 100: [: # mark_description : integer expression expected when using Intel 2024 HPC compilers

Guessing I am doing something wrong or perhaps OpenBLAS won't compile with 2024 Intel compiler?

`[feacluster@intel OpenBLAS]$ make CC=icx FC=ifort INTERFACE64=1 BINARY=64 USE_THREAD=0 USE_LOCKING=0 PREFIX=/home/feacluster/OpenBLAS/build
./f_check: line 100: [: # mark_description : integer expression expected
getarch_2nd.c:22:35: error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_M'
   22 |     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
      |                                   ^                       ^
.... snipped`

--- a/cpuid_x86.c
+++ b/cpuid_x86.c
@@ -1535,6 +1535,7 @@ int get_cpuname(void){
               return CPUTYPE_SANDYBRIDGE;
             else
               return CPUTYPE_NEHALEM;
+	  case 0: // Meteor Lake
           case 7: // Rocket Lake           
 	    if(support_avx512())
               return CPUTYPE_SKYLAKEX;
@@ -1560,6 +1561,19 @@ int get_cpuname(void){
 	      return CPUTYPE_NEHALEM;
         }
         break;
+      case 12: //family 6 exmodel 12
+	switch (model) {
+	  case 15:
+	    if(support_avx512())
+              return CPUTYPE_SAPPHIRERAPIDS;
+            if(support_avx2())
+              return CPUTYPE_HASWELL;
+            if(support_avx())
+	      return CPUTYPE_SANDYBRIDGE;
+	    else
+	    return CPUTYPE_NEHALEM;
+	  }
+	break;
       }
       break;    
     case 0x7:

